Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm

```diff
@@ -1660,6 +1658,14 @@ void getSupportedTypes(HashSet<String>& types) const final
     if (m_videoFrameMetadataGatheringObserver)
         return;
     m_isGatheringVideoFrameMetadata = true;
+    if (RefPtr videoRenderer = layerOrVideoRenderer()) {
+        videoRenderer->notifyWhenHasAvailableVideoFrame([weakThis = WeakPtr { *this }](const MediaTime& presentationTime, double displayTime) {
+            if (RefPtr protectedThis = weakThis.get()) {
+                protectedThis->setHasAvailableVideoFrame(true);
```

Comment:
I am not sure why this is needed here.

Reply:
I didn't have it at first, causes plenty of rVFC test failures otherwise. The call to `setHasAvailableVideoFrame` must be made whenever there's a a new one.
I wasn't going to sort that out and causes unexpected behaviour changes

Reply:
ok, I identified the problem. Now we can properly have two separate callbacks.